sotu.md

```diff
@@ -1193,61 +1193,49 @@ overload, such as:
 
 ## Testing
 
-**Rating:** Mature
-
-There are a few places where Haskell is the clear leader among all languages:
+**Rating:** Best in Class
 
-* property-based testing
-* mocking / dependency injection
+In terms of testing, Haskell is a clear leader.
+It has both more practical, more advanced and more advanced test automation than other languages.
 
-Haskell's `QuickCheck` is the gold standard which all other property-based
-testing libraries are measured against.  The reason `QuickCheck` works so
-smoothly in Haskell is due to Haskell's type class system and purity.  The type
-class system simplifies automatic generation of random data from the input type
-of the property test.  Purity means that any failing test result can be
+Haskell's property testing, which started with `QuickCheck`, is the gold
+standard which all other property-based testing implementations are measured
+against.  The reason property testing works so smoothly in Haskell is due to
+Haskell's strong static types and type class system.  The strong static types
+allow for automatic generation of what would otherwise be boiler plate;
+generators, shrinking, general properties.  The type class system simplifies
+how tests are defined.  Shrinking means that any failing test result can be
 automatically minimized by rerunning the check on smaller and smaller inputs
-until `QuickCheck` identifies the corner case that triggers the failure.
-
-Mocking is another area where Haskell shines because you can overload almost
```

Comment:
I removed the section on mocking as it is a Bad Idea (but I recognise that this is an opinionated position so I'm happy to put it back).
Instead I added a bit more about integration testing.

Reply:
You might want at least to add mocking as an available choice in the list of expected functionalities, outlined right after this paragraph